### PR TITLE
A tiny bug fix: Change infosAsSlice() to call removeInternal()

### PR DESCRIPTION
**Pull request by kkaneda**
`removeInternal()` updates a gatekeeper while `delete()` doesn't. This commit also makes cosmetic changes to call `expired()` whenever possible.

### CODE REVIEW

File: gossip/group.go [hunk 1]
```diff
@@ -124,7 +124,10 @@ func (g *group) compact() bool {
 	g.minTTLStamp = math.MaxInt64
 	g.gatekeeper = nil
 	for key, i := range g.Infos {
-		if i.TTLStamp <= now {
+		if i.expired(now) {
+			// We do not need to call g.removeInternal()
+			// since we will soon add a new info that
+			// becomes a new gatekeeper.
```
Comment (petermattis) [inline gossip/group.go]: I'd rephrase as: `We do not need to call g.removeInternal() since we've already cleared the gatekeeper and minTTLStamp`. That said, I don't think there is any harm in calling `removeInternal` here and doing so localizes the deletion from the `Infos` map to `removeInternal`.

**Comment by petermattis** [comment]
LGTM

**Comment by kkaneda** [comment]
Thanks for the review, Peter. Agreed. I've just replaced `delete` with `removeInternal`.

**Comment by petermattis** [comment]
LGTM
